server/test/views/components/ModalTest.java

```diff
@@ -68,4 +79,78 @@ public void buildsModal_withRepeatOpenBehavior_onlyShowOnceWithBypassUrl() {
     assertThat(rendered).contains("only-show-once-group=\"programs_index_login_prompt\"");
     assertThat(rendered).contains("bypass-url=\"https://bypassurl.com\"");
   }
+
+  @Test
+  public void buildsModal_adminFacingNoMessages_usesDefaultClose() {
+    Modal modal =
+        Modal.builder()
+            .setModalId("modal-id")
+            .setLocation(Modal.Location.ADMIN_FACING)
+            .setContent(div().with(text("content")))
+            .setModalTitle("Modal Title")
+            .build();
+    String rendered = modal.getContainerTag().render();
+
+    assertThat(rendered).contains("aria-label=\"Close\"");
+  }
+
+  @Test
+  public void buildsModal_debugNoMessages_usesDefaultClose() {
+    Modal modal =
+        Modal.builder()
+            .setModalId("modal-id")
+            .setLocation(Modal.Location.DEBUG)
+            .setContent(div().with(text("content")))
+            .setModalTitle("Modal Title")
+            .build();
+    String rendered = modal.getContainerTag().render();
+
+    assertThat(rendered).contains("aria-label=\"Close\"");
+  }
+
+  @Test
+  public void buildsModal_applicantFacingNoMessages_throwsException() {
+
+    Modal.Builder modalBuilder =
+        Modal.builder()
+            .setModalId("modal-id")
+            .setLocation(Modal.Location.APPLICANT_FACING)
+            .setContent(div().with(text("content")))
+            .setModalTitle("Modal Title");
+
+    assertThatThrownBy(modalBuilder::build).isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void buildsModal_applicantFacingHasMessages_usesCloseFromMessage() {
+    Lang lang = Lang.forCode("es-US");
+
+    // Note: There's a bit of an API mismatch in Play, where [stubMessagesApi] expects
+    // classes in the play.i18n package but [stubLangs] expects classes in the
+    // play.api.i18n package, which is why the same Lang is created twice.
```

Comment:
That's unfortunate! I'm was also surprised to learn that we've never tried to mock the messages themselves so I'm glad you figured this out.

Reply:
Yeah, it's a bit clunky but does seem to work. If we need to do more of it elsewhere we  can make a utility method at that point.